src/borg/testsuite/helpers.py

```diff
@@ -65,6 +65,24 @@ def test_ssh(self, monkeypatch):
             "Location(proto='ssh', user='user', host='::', port=1234, path='/some/path', archive=None)"
         assert repr(Location('ssh://user@[::]/some/path')) == \
             "Location(proto='ssh', user='user', host='::', port=None, path='/some/path', archive=None)"
+        assert repr(Location('ssh://user@[2001:db8::]:1234/some/path::archive')) == \
+            "Location(proto='ssh', user='user', host='2001:db8::', port=1234, path='/some/path', archive='archive')"
```

Comment:
hmm, guess our ssh code reassembles this like `ssh -p port user@host ...`, so if the ipv6 addr has no [] any more, does this actually work?